config/instance/registry_metadata_report.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package instance
+
+import (
+	"sync"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common"
+	"dubbo.apache.org/dubbo-go/v3/common/extension"
+	"dubbo.apache.org/dubbo-go/v3/metadata/report"
+)
+
+var (
+	regInstances = make(map[string]report.MetadataReport, 8)
+	mux          sync.RWMutex
+)
+
+// GetMetadataReportInstanceByReg obtain metadata report instance through registry config
+func GetMetadataReportInstanceByReg(protocol string) report.MetadataReport {
```

Comment:
Is `GetMetadataReportInstanceByProtocol` better than current name?

Reply:
GetMetadataReportInstanceByRegistryProtocol is better, but it's too long. 
Can discuss

Reply:
too naive.  ’GetMetadataReportInstance(RegistryProtocol string)' is enough.

Reply:
> 

Big Old, there's a duplicate name

Reply:
Then I think the suggestion from @justxuewei is better.

Reply:
In most cases, "reg" means regular expression, and it rarely stands for "registry". IMO,  both "Protocol" or "RegistryProtocol" sound good to me.

Reply:
Change to GetMetadataReportByRegistyProtocol